protocols/2a4397/ngs_library_prep.ot2.py

```diff
@@ -0,0 +1,216 @@
+
+    # pipettes
+    p10 = instruments.P10_Single(mount=p10_mount, tip_racks=tips10)
+    p300 = instruments.P300_Single(mount=p300_mount, tip_racks=tips300)
+
+    # setup
+    original_samples = original_sample_plate.wells()[:number_of_samples]
+    reactions = rxn_plate.wells()[:number_of_samples]
+    barcodes = barcode_plate.wells(
+        barcode_start_well, length=number_of_samples)
+    mag_samples = magplate.wells()[:number_of_samples]
+
+    tip10_count = 0
+    tip10_max = 96*len(tips10)
+    tip300_count = 0
+    tip300_max = 96*len(tips300)
+
+    def tip_check(pip):
+        nonlocal tip10_count
+        nonlocal tip300_count
+        if pip == 'p10':
+            if tip10_count == tip10_max:
+                robot.pause('Replace 10ul tipracks before resuming.')
+                tip10_count = 0
+                p10.reset()
+            tip10_count += 1
+        elif pip == 'p300':
+            if tip300_count == tip300_max:
+                robot.pause('Replace 300ul tipracks before resuming.')
+                tip300_count = 0
+                p300.reset()
+            tip300_count += 1
+
+    robot.comment('Transfer 10ng genomic DNA into the sample plate. Adjust the \
+volume to 10ul with nuclease-free water. Mix thoroughly by flicking the tube. \
+Spin down briefly in a microfuge, and mount in slot 4.')
+
+    # prepare reaction mix
+    for o, r in zip(original_samples, reactions):
+        tip_check('p10')
+        p10.transfer(25, longamp_mm, r)
+        tip_check('p10')
+        p10.pick_up_tip()
+        p10.transfer(5, nuc_free_water, r, new_tip='never')
+        p10.transfer(10, o, r, new_tip='never')
+        p10.drop_tip()
+
+    robot.pause('Mix gently by flicking the tube, and spin down. Remove the \
+foil of the barcode plate for pipette access.')
+
+    # add barcode
+    for b, r, in zip(barcodes, reactions):
+        tip_check('p10')
+        p10.pick_up_tip()
+        p10.transfer(10, b, r, new_tip='never')
+        p10.mix(5, 9, r)
+        p10.blow_out(r.top())
+        p10.drop_tip()
+
+    robot.pause('Thermocycle according to parameters prescribed in kit \
+manual. Replace on slot 1 when finished.')
+
+    # transfer DNA samples, add beads, and mix
+    tip_check('p300')
+    p300.pick_up_tip()
+    p300.mix(10, 200, ampure_xp_beads)
+    p300.blow_out(ampure_xp_beads.top())
+    for i, (r, m) in enumerate(zip(reactions, mag_samples)):
+        if not p300.tip_attached:
+            tip_check('p300')
+            p300.pick_up_tip()
+        p300.transfer(30, ampure_xp_beads, m, new_tip='never')
+        p300.transfer(50, r, m, new_tip='never')
+        p300.mix(5, 30, m)
+        p300.blow_out(m.top())
+        p300.drop_tip()
+
+    robot.pause('Incubate on a Hula mixer (rotator mixer) for 5 minutes at \
+RT. Then spin down the samples and place on the magnetic module.')
+
+    robot._driver.run_flag.wait()
+    magdeck.engage(height=18)
+
```

Comment:
Should there be a delay to allow beads to settle before the wash?

Reply:
good catch. thanks